### PR TITLE
[wasm][debugger] Trying to fix firefox flaky tests on CI.

### PR DESCRIPTION
**Pull request by thaystg**
Fix Firefox tests on CI.

### CODE REVIEW

File: src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestFirefox.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public override async Task InitializeAsync()
             };
 
         await Ready();
-        await insp.OpenSessionAsync(fn, TestTimeout);
+        await insp.OpenSessionAsync(fn, TestTimeout*4);
```
Comment (ilonatommy) [inline src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestFirefox.cs:40]: Can we save this info as a fixed timeout for Firefox? I mean: 

```
public class DebuggerTestBase
{
    private const int DefaultTestTimeoutMs = 1 * 60 * 1000;
    protected TimeSpan TestTimeout => TimeSpan.FromMilliseconds(DefaultTestTimeoutMs);
}

public class DebuggerTestFirefox : DebuggerTestBase
{
    private new TimeSpan TestTimeout => base.TestTimeout * 4;
}
```

```suggestion
        await insp.OpenSessionAsync(fn, TestTimeout);
```

File: src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs [hunk 2]
```diff
@@ -71,38 +71,49 @@ protected override async Task<WasmDebuggerConnection> SetupConnection(Uri webser
             throw new Exception($"Failed to connect to the proxy at {endpoint}", se);
         }
     }
+    public async Task<bool> ProcessTabInfo(Result command, CancellationToken token)
+    {
+        if (command.Value?["result"]?["value"]?["tabs"] == null ||
+            command.Value?["result"]?["value"]?["tabs"]?.Value<JArray>()?.Count == 0 ||
+            command.Value?["result"]?["value"]?["tabs"]?[0]?["url"]?.Value<string>()?.StartsWith("about:") == true)
```
Comment (radical) [inline src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs:78]: Extract to a local - `command.Value?["result"]?["value"]?["tabs"]`

Comment (radical) [inline src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs:78]: `StartsWith` - use `OrdinalIgnoreCase` with it.

File: src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs [hunk 3]
```diff
@@ -71,38 +71,49 @@ protected override async Task<WasmDebuggerConnection> SetupConnection(Uri webser
...
+        BreakpointActorId = res.Value?["result"]?["value"]?["breakpointList"]?["actor"]?.Value<string>();
+        return true;
+    }
 
     public override async Task ProcessCommand(Result command, CancellationToken token)
     {
-        if (command.Value?["result"]?["value"]?["tabs"] != null)
+        if (await ProcessTabInfo(command, token))
+            return;
+        do
```
Comment (radical) [inline src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs:113]:
```suggestion
        do
```

File: src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs [hunk 4]
```diff
@@ -71,38 +71,49 @@ protected override async Task<WasmDebuggerConnection> SetupConnection(Uri webser
             throw new Exception($"Failed to connect to the proxy at {endpoint}", se);
         }
     }
+    public async Task<bool> ProcessTabInfo(Result command, CancellationToken token)
```
Comment (radical) [inline src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs:74]: What is this method doing?

Comment (ilonatommy) [reply src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs:74]: @thaystg, I was trying to understand the general meaning but failed, could you help here?

**Review by radical — APPROVED** [review]
As long as the chrome, and firefox tests pass, LGTM 👍

**Comment by radical** [comment]
Which tests is this intended to fix? There are still some failures, but those might be new: https://dev.azure.com/dnceng-public/public/_build/results?buildId=112393&view=logs&jobId=cd9e019e-1e38-5f0c-5d75-21696d4ad4f5&j=cd9e019e-1e38-5f0c-5d75-21696d4ad4f5&t=d390e7cf-32c2-5383-2406-50456656d9e1

File: src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs (hunk 4 above)
Comment (thaystg) [reply src/mono/wasm/debugger/DebuggerTestSuite/FirefoxInspectorClient.cs:74]: Getting the active tab to attach the debugger.

**Pull request merged**